app/src/test/java/org/simple/clinic/security/pin/PinEntryCardControllerTest.kt

```diff
@@ -68,75 +65,83 @@ class  PinEntryCardControllerTest {
 
   @Test
   fun `when 4 digits are entered then the PIN should be submitted automatically`() {
+    val pinDigest = "1234"
     uiEvents.onNext(PinTextChanged("1"))
     uiEvents.onNext(PinTextChanged("12"))
     uiEvents.onNext(PinTextChanged("123"))
     uiEvents.onNext(PinTextChanged("1234"))
+    uiEvents.onNext(PinDigestToVerify(pinDigest))
 
-    verify(passwordHasher, times(1)).compare(loggedInUser.pinDigest, "1234")
+    verify(passwordHasher, times(1)).compare(pinDigest, "1234")
```

Comment:
We can remove explicit `times(1)` here?

Reply:
Also, why do we have the variable `pinDigest` and not using it here `.compare(pinDigest, pinDigest)`

Reply:
What this test is actually testing is that "when four digits are entered, the value should be compared to the PIN digest that was set".

This might seem a little confusing because both the value of `pinDigest` and the PIN text changed value that is emitted are set to the same value "1234".

Reply:
> We can remove explicit `times(1)` here?

Done.